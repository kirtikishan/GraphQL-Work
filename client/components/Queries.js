import gql from 'graphql-tag';


export const homePageQuery = gql`
  {
     carOfTheWeek{
          modelId,
          review,
          carModel {
            makeId,
            name,
            price,
            imageUrl
            make {
              name
            }
          }
     }
  }
`;

export const searchQuery = gql`
      query search {
         makes {
            id,
            name
            models {
                id,
                name
            }
         }
      }
    `;


export const detailsQuery = gql`
  query ModelsQuery($id : Int!) {
    model(id: $id) {
      name,
      price,
      imageUrl,
      make {
          id,
          name
      }
    }
  }
`;