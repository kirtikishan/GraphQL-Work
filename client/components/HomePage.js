import React, {Component} from 'react';
import gql from 'graphql-tag';
import {graphql} from 'react-apollo';
import BEMHelper from 'react-bem-helper';
import {homePageQuery} from './Queries';
var homePage = new BEMHelper('home-page');
var homePageDetails = new BEMHelper('home-page__details');
export class HomePage extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        let {carOfTheWeek, loading} = this.props.data;

        if(loading) {
            return (<div className="loading"></div>);
        } else {
            return (
                <div {...homePage()}>
                    <span {...homePage('headline')}>car of the week</span>
                    <div {...homePage('row1')}>
                        <span {...homePage('make')}>{carOfTheWeek.carModel.make.name}</span>
                        <span {...homePage('name')}>{carOfTheWeek.carModel.name}</span>
                    </div>
                    <img {...homePage('banner')}
                         src={carOfTheWeek.carModel.imageUrl}></img>
                    <div {...homePageDetails('')}>
                        <div {...homePageDetails('row2')}>
                            <span {...homePageDetails('review')}>{carOfTheWeek.review}</span>
                        </div>
                        <div {...homePageDetails('row3')}>
                            <span {...homePageDetails('text')}>drive away price</span>
                            <span {...homePageDetails('price')}> $ {carOfTheWeek.carModel.price}</span>
                        </div>
                    </div>
                </div>
            );
        }
    }
}

export default graphql(homePageQuery)(HomePage);
