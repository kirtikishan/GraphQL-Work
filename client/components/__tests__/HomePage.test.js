import * as React from 'react';
import {HomePage} from '../HomePage';
import {shallow} from 'enzyme';
import {homePageQuery} from '../Queries';

describe('HomePage Component', () => {
    it('should render the homepage correctly', () => {
        const data = {
            carOfTheWeek: {
                modelId: 520,
                review: "The Mazda MX-5 is a traditional two-seat sports car, with a lightweight body and rear-wheel drive. It has a folding, fabric roof and is among the least expensive convertibles. This fourth-generation MX-5 is fantastic fun to drive. Motoring magazine Wheels named it Car of the Year for 2016.",
                carModel: {
                    makeId: 50,
                    name: "MX-5",
                    price: 90000,
                    imageUrl: "https://www.mazda.com.au/globalassets/settings/vehicle-assets/mx-5/2017-01-ipm/side-and-rear-34/mx-5_softtop_soulred_buildbody_front.png",
                    make: {
                        name: "Mazda"
                    }
                }
            }

        }

        const wrapper = shallow(
            <HomePage data={data}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should be rendered the correct home page query', () => {
        expect(homePageQuery).toMatchSnapshot();
    });
})
;
