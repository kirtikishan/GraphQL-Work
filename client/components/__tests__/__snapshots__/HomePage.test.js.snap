// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should be rendered the correct home page query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": null,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "carOfTheWeek",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "modelId",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "review",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "carModel",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "makeId",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "price",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "imageUrl",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "make",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": null,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": null,
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 235,
    "start": 0,
  },
}
`;

exports[`HomePage Component should render the homepage correctly 1`] = `
<div
  className="home-page"
>
  <span
    className="home-page__headline"
  >
    car of the week
  </span>
  <div
    className="home-page__row1"
  >
    <span
      className="home-page__make"
    >
      Mazda
    </span>
    <span
      className="home-page__name"
    >
      MX-5
    </span>
  </div>
  <img
    className="home-page__banner"
    src="https://www.mazda.com.au/globalassets/settings/vehicle-assets/mx-5/2017-01-ipm/side-and-rear-34/mx-5_softtop_soulred_buildbody_front.png"
  />
  <div
    className="home-page__details"
  >
    <div
      className="home-page__details__row2"
    >
      <span
        className="home-page__details__review"
      >
        The Mazda MX-5 is a traditional two-seat sports car, with a lightweight body and rear-wheel drive. It has a folding, fabric roof and is among the least expensive convertibles. This fourth-generation MX-5 is fantastic fun to drive. Motoring magazine Wheels named it Car of the Year for 2016.
      </span>
    </div>
    <div
      className="home-page__details__row3"
    >
      <span
        className="home-page__details__text"
      >
        drive away price
      </span>
      <span
        className="home-page__details__price"
      >
         $ 
        90000
      </span>
    </div>
  </div>
</div>
`;
